### ISSUE TITLE
From media library deleted images are still visible in code editor

### ISSUE THREAD

**Issue by sommer-m** [WordPress/gutenberg#11501]
**Describe the bug**
If an image is referenced in article and then delete this from the library, the image block is still visible in the editor with "Code Editor" mode.

**To Reproduce**
1. Create an article with an image.
2. Publish the article.
3. Delete the image in the library. 
4. Go back to the article. The image is not shown in the "Visual Editor" mode.
5. Then change the mode to "Code Editor". The image block is still shown. 
6. Add a new image (editor in "Visual Editor" mode). 
7. Open Preview of the article. Sometimes there is no image or the old image is shown. 

**Expected behavior**
The image block in "Code Editor" mode is deleted and the preview shows the correct image.

**Desktop (please complete the following information):**
 - OS: MacOS 10.13.6
 - Browser: Chrome Version 69.0.3497.100 

**Additional context**
- Gutenberg Version 4.0.0

**Comment by designsimply** [comment]
This is expected. If you reference an image (either from the media library on your own blog or elsewhere on the internet), the HTML pointing to the image should remain even if the image is removed. 

Let's imagine another scenario. Let's say you added an image of a logo to your media library. Then let's say you reference that logo 100 times in various posts and pages and widgets over time. If you delete the logo file from your media library after that, the expectation is not that all of the 100 different places that the HTML referenced that image is removed. Instead, either the image is simply not displayed or a broken image is displayed. I could see making a request to ask that a broken image be displayed in the editor for a case like this, would that make sense to you as an enhancement request?

I tested just now with Chrome 70.0.3538.77, Firefox 63.0, and Safari 12.0 on macOS 10.13.6 and Safari was the only browser that displayed a broken image on the front end for a case where an image is referenced in the HTML but then deleted. ([screenshot](https://user-images.githubusercontent.com/1119271/48030785-629cc380-e10f-11e8-9ea8-2a7d761973f6.png))

**Comment by sommer-m** [comment]
Thank you for your prompt reply.
In my opinion, it makes sense that a broken image in the editor is displayed, mayby with an info box beside it "Your picture has been removed".
It would also be nice if a message or dialog is displayed when deleting a referenced image in the media library and the user can choose to delete that image or not.

**Comment by paaljoachim** [comment]
Hi

I am testing with:
WordPress version 5.7 beta 1.
Theme Twenty Twenty One.
Gutenberg plugin: 9.9.1
Chrome
A local site
OSX 10.15.7

Testing instructions:

1. Create an article with an image.
2. Publish the article.
3. Delete the image in the library.
4. Go back to the article. The image is not shown in the "Visual Editor" mode.
5. Then change the mode to "Code Editor". The image block is still shown.
6. Add a new image (editor in "Visual Editor" mode).
7. Open Preview of the article. Sometimes there is no image or the old image is shown.


I made a post with an image. Published the post. Went to the Media Library and deleted the image.
Went back to the article/post. I am still seeing the image in the Image block. (Even though it should have been deleted.)

I opened up the Code Editor and could also see the image code.
Went back to the Visual Editor.

Clicked the Image block containing the deleted image that is still seen. Clicked the Replace button.
Image is replaced with another. Clicked Preview and noticed the new image.

My conclusion is that the issue created is not valid any longer.

-----

I am wondering though....
if we should have a placeholder screen for media (images, audio etc) that have been deleted from the Media Library.
